src/string/VarList.cpp

```diff
@@ -5,25 +5,54 @@
 
 using namespace Hyprutils::String;
 
-Hyprutils::String::CVarList::CVarList(const std::string& in, const size_t lastArgNo, const char delim, const bool removeEmpty) {
-    if (!removeEmpty && in.empty())
+// Original constructor calls the extended one with handleEscape = false
+Hyprutils::String::CVarList::CVarList(const std::string& in, const size_t lastArgNo, const char delim, const bool removeEmpty) :
+    CVarList(in, lastArgNo, delim, removeEmpty, false) {}
+
+// Extended constructor with escape handling parameter
+Hyprutils::String::CVarList::CVarList(const std::string& in, const size_t lastArgNo, const char delim, const bool removeEmpty, const bool handleEscape) {
+    if (!removeEmpty && in.empty()) {
         m_vArgs.emplace_back("");
+        return;
+    }
+
+    std::string currentArg;
+    bool        escaped = false;
+    size_t      idx     = 0;
 
-    std::string args{in};
-    size_t      idx = 0;
-    size_t      pos = 0;
-    std::ranges::replace_if(args, [&](const char& c) { return delim == 's' ? std::isspace(c) : c == delim; }, 0);
-
-    for (const auto& s : args | std::views::split(0)) {
-        if (removeEmpty && s.empty())
-            continue;
-        if (++idx == lastArgNo) {
-            m_vArgs.emplace_back(trim(in.substr(pos)));
-            break;
+    for (size_t i = 0; i < in.length(); ++i) {
+
+        // Handle escape character if enabled
+        if (handleEscape && in[i] == '\\') {
+            escaped = true;
+            i++;
         }
-        pos += s.size() + 1;
-        m_vArgs.emplace_back(trim(s.data()));
+
+        char c = in[i];
+
+        // Determine if this char is a delimiter (respect escape setting)
+        const bool isDelim = (delim == 's' ? std::isspace(c) : c == delim) && !(handleEscape && escaped);
+
+        if (isDelim) {
+            if (!removeEmpty || !currentArg.empty()) {
+                m_vArgs.emplace_back(trim(currentArg));
+                currentArg.clear();
+                idx++;
+            }
+
+            if (idx == lastArgNo - 1) {
+                m_vArgs.emplace_back(trim(in.substr(i + 1)));
+                return;
+            }
+        } else
+            currentArg += c;
```

Comment:
don't do this. This is extremely slow and causes a reallocation galore. Instead, store indices, then substr and replace.